### PR TITLE
updates to Nodesobject so it fits reacts needs

### PR DESCRIPTION
**Pull request by boekhorstb1**
Probably this needs to be adapted:
- action
- algorithm could probably be made more efficient

### CODE REVIEW

File: lib/Tree/Nodesobject.php [hunk 1]
```diff
@@ -23,13 +23,60 @@ class Passwd_Tree_Nodesobject extends Horde_Tree_Renderer_Base
...
+
+            if (isset($value['children'])) {
+                # if children, then assing parent to child
+                foreach($value['children'] as $int => $childname){
+                    foreach($reactOb as $reactkey => $reactvalue){
+                        if($reactkey === $childname){
+                            $reactOb[$reactkey]['parent'] = $key;
+                        }
+                    }
+                }
```
Comment (midahp) [inline lib/Tree/Nodesobject.php]: You can simplify this like so:
```suggestion
                foreach($value['children'] as $childname){
                    $reactOb[$childname]['parent'] = $key;
                }
```

File: lib/Tree/Nodesobject.php [hunk 2]
```diff
@@ -23,13 +23,60 @@ class Passwd_Tree_Nodesobject extends Horde_Tree_Renderer_Base
...
+        // Resulting object for react
+        $reactOb = [];
+
+        // Creating basic array without parents
+        foreach($nodes as $key => $value){
+
+            $reactOb_Value['entryId'] = $key;
+            $reactOb_Value['type'] = 'entry';  
+            $reactOb_Value['parent'] = '';
+            $reactOb_Value['caption'] = $key;
```
Comment (midahp) [inline lib/Tree/Nodesobject.php]: Caption should be the `value['label']` of the original array

File: lib/Tree/Nodesobject.php (hunk 1 above)
Comment (boekhorstb1) [reply lib/Tree/Nodesobject.php]: o yes! So true. Will adapt.

File: lib/Tree/Nodesobject.php (hunk 2 above)
Comment (boekhorstb1) [reply lib/Tree/Nodesobject.php]: ok will change this

**Comment by boekhorstb1** [comment]
I updated the code, see commits tab ([here](https://github.com/midahp/passwd/pull/2/commits))

**Pull request merged**